app/core/utils.py

```diff
@@ -198,6 +204,128 @@ def get_gpu_info():
     return gpu_name, total_memory_value, used_memory_value, mem_percentage
 
 
+def extract_labels_from_html_file(html_content, class_names):
+    soup = BeautifulSoup(html_content, "html.parser")
+    labels = soup.select(class_names)
+    return (label.get_text() for label in labels)
+
+
+def find_maximum_label(labels):
+    pattern = re.compile(r"v\d+\.\d+\.\d+$")
+    return max(filter(pattern.match, labels), default=None, key=parse_version)
+
+
+def get_premd_last_tag(owner, repository, package):
+    response = requests.get(
+        f"https://github.com/{owner}/{repository}/pkgs/container/{package}"
```

Comment:
Why not calling the API instead?
Something like this, with an authentication token if necessary:
```
def check_premd_update():
    try:
        url = f'https://api.github.com/orgs/premai-io/packages/container/premd/versions'
        headers = {
            'Accept': 'application/vnd.github+json'
        }
        response = requests.get(url, headers)
        if response.status_code == 200:
            data = response.json()
            if data:
                # Find the latest version by sorting by created_at in descending order
                latest_version = max(data, key=lambda x: x['created_at'])
                return latest_version['name']
            else:
                print("No versions found for the image.")
        else:
            logger.error(f"Failed to retrieve data from GitHub API. Status code: {response.status_code}")
            return None
    except Exception as e:
        logger.error(f"An error occurred: {e}")
        return None
```

Reply:
because we'd have to hardcode an auth token ([this particular API requires authentication even to access public info](https://docs.github.com/en/rest/packages/packages))